index.rst

```diff
@@ -96,6 +98,8 @@ DatasetMetatype
 
 A category of :ref:`DatasetTypes <DatasetType>` that utilize the same in-memory classes for their :ref:`ConcreteDatasets <ConcreteDataset>` and can be saved to the same file format(s).
 
+[KSK] as above, calling it a ``Metatype`` seems to hide the fact that they are concrete implementations.  Mappings of ``DatasetHandle``\ s to ``DatasetType``\ s that, when constructed, result in ``ConcreteDataset``\ s, is a more intutive treatment to me.
```

Comment:
I don't quite see how it's a mapping (maybe clarifying what a `DatasetHandle` is would help?)

The existing nomenclature comes from the idea that you "instantiate" a `DatasetMetatype` when you define a `DatasetType`, and you "instantiate" a `DatasetType` when you write a `Dataset`.  Maybe it'd be better to go with an inheritance analogy for one of these layers, though and say that you e.g. "specialize" an `AbstractDatasetType` when you define a `DatasetType` (but then we'd probably have to rename `ConcreteDataset`).  But I am also curious to hear your mapping nomenclature more spelled out.

Reply:
Here `DatasetHandle` means `DatasetType` in your terminology.

It may not be useful but I was thinking literally of a mapping like:
```type_map = {lsst.afw.Exposure:('calexp', 'coadd', 'post_isr'), lsst.afw.SourceCatalog:('src', 'ic_src', 'deep_src')}```.
One would want to reverse the mapping, but this direction fits your words.  I.e. for a `DatasetType` in one of those tuples, you get a `ConcreteDataset` of that type.  I'm not sure this distinction is important.